lib/engine/round/g_1822/stock.rb

```diff
@@ -0,0 +1,76 @@
+# frozen_string_literal: true
+
+require_relative '../stock'
+
+module Engine
+  module Round
+    module G1822
+      class Stock < Stock
+        attr_accessor :bids, :bidders
+
+        def finish_round
+          @game.bidbox_minors.map do |minor_company|
+            bid = highest_bid(minor_company)
+            float_minor(bid) unless bid.nil?
+          end
+
+          @game.bidbox_concessions.map do |concessions|
+            bid = highest_bid(concessions)
+            buy_company(bid) unless bid.nil?
+          end
+
+          @game.bidbox_privates.map do |private_company|
+            bid = highest_bid(private_company)
+            buy_company(bid) unless bid.nil?
+          end
+
+          super
+        end
+
+        def setup
+          super
+        end
+
+        def buy_company(bid)
+          player = bid.entity
+          company = bid.company
+          price = bid.price
+
+          company.owner = player
+          player.companies << company
+          player.spend(price, @game.bank) if price.positive?
+          @game.companies.delete(company)
+          @log << "#{player.name} buys #{company.name} for #{@game.format_currency(price)}"
+        end
+
+        def float_minor(bid)
+          player = bid.entity
+          company = bid.company
+          price = bid.price
+
+          return unless (minor = @game.minors.find { |m| m.id == company.id })
```

Comment:
You've detected a broken game here. Raise an error to reject the last action rather than silently continue on.

Reply:
In general, raise an error when something happens that shouldn't be possible.

Reply:
True will fix, shouldnt need to do the return.